### DISCUSSION TITLE
v7.0.0-b  - pre-release

### DISCUSSION THREAD

**Discussion by marksie1988** [totaldebug/atomic-calendar-revive#566] [Announcements]
Hello Everyone,

I have created a new pre-release v7.0.0-b which sees quite a large rewrite along with some long awaited features that I was having difficulty implementing (hence the rewrite)

It would be great if some of you could install this version and let me know if you see any issues!

Please make sure you review your configuration as there are some breaking changes in there.

[v7.0.0-b](https://github.com/totaldebug/atomic-calendar-revive/releases/tag/v7.0.0-b)

- feat: option showCalendarName to display the calendar name on the event view
- feat: option showMultiDayEventParts to split multi day events per day
- feat: option showWeekNumber to display the week number at the top of each week
- feat: ability to add entities via the visual editor
- fix: issue where HTML descriptions were not displayed correctly
- fix: incorrect links in documentation
- fix: resolves issues with using CAST 
- docs: migrated documentation to sphinx for better user experience

## :boom: Breaking Changes

This version includes some breaking changes, you must check your configuration after upgrading:

- blacklist renamed to blocklist
- whitelist renamed to allowlist
- locationWhitelist changed to allowlistLocation
- added blocklistLocation
- Blocklist and allowlist now changed to use Regex for more flexibility
- showDeclined changed to hideDeclined
- eventCalName changed to name under each entity

**Comment by BebeMischa** [comment]
Installed this pre-release yesterday. Went smooth on first sight.
Today I came back from work, all my Lovelace cards containing media players went away. Just completely disappeared.
After downgrade to latest stable ACR they came back immediately.

**Reply by marksie1988** [reply]
Thanks for the feedback, I think I know the cause of that and will look into a fix.

**Reply by marksie1988** [reply]
This has now been fixed in v7.0.0-b1

**Reply by BebeMischa** [reply]
I will try tomorrow ;-)

**Comment by WarC0zes** [comment]
I tried version 7.0.0b, which improves a lot of things. But I noticed that it requires a little more resource from the machine. I'm on rpi3 1go I'm running on average at 66% ram usage and with version 7.0.0b I'm at 72% on average with peaks at 93%. I downgrade while waiting because my rpi3 suffers a little.
Thank for you works.

**Reply by marksie1988** [reply]
This is interesting, Javascript is Client Side, which means all the processing is done in the client rather than on the server, im not actually sure how the card would be causing the memory usage to increase, I will have to look into this some.

**Reply by WarC0zes** [reply]
Hi, 
I found the ram consumption problem, it was due to core 2022.3. I updated to 2022.3.2 and installed 7.0.0b2, I no longer he ram consumption problem. So it's not your card to blame ;)

**Comment by floriangeigl** [comment]
Hi - any idea why I can't find this integration in HACs anymore? Had it installed, but it stopped working some weeks ago. Since, then I'm not able to find it anymore.
![image](https://user-images.githubusercontent.com/2580073/158013932-97105390-3a43-40f8-9c3f-894ac35580e4.png)
![image](https://user-images.githubusercontent.com/2580073/158013943-019d7f8c-0913-486f-b04c-103ca9d55ab6.png)

**Reply by marksie1988** [reply]
Hi, I have just checked on my instance and it's still coming up as expected. If you already have the card showing in the main screen you may have to choose the option to redownload, or remove it and search again. 

![Screenshot_20220312-121823](https://user-images.githubusercontent.com/159166/158017671-5185a138-1777-4e24-a60c-4e43a45ab495.jpg)

**Reply by floriangeigl** [reply]
thx - re-downloading fixed the issue.
